libraries/standard/mqtt/src/mqtt.c

```diff
@@ -904,6 +916,8 @@ static MQTTStatus_t handleIncomingAck( MQTTContext_t * pContext,
     assert( pContext->appCallback != NULL );
 
     appCallback = pContext->appCallback;
+    /* We did not receive a publish. */
+    deserializedInfo.pPublishInfo = NULL;
```

Comment:
This logic can be moved to the if-block after line 961 for a better logical placement.

Reply:
I mainly put this out here since to reduce complexity since this will always be NULL, but the complexity of this function is 5 so it doesn't really matter. Moved to the if